lightning_pose/data/utils.py

```diff
@@ -393,9 +440,9 @@ def evaluate_heatmaps_at_location(
 
 @typechecked
 def undo_affine_transform(
-        keypoints: TensorType["seq_len", "n_keypoints", 2],
-        transform: Union[TensorType["seq_len", 2, 3], TensorType[2, 3]]
-) -> TensorType["seq_len", "n_keypoints", 2]:
+    keypoints: TensorType["seq_len", "num_keypoints", 2],
+    transform: Union[TensorType["seq_len", 2, 3], TensorType[2, 3]]
+) -> TensorType["seq_len", "num_keypoints", 2]:
 
     # add 1s to get keypoints in projective geometry coords
     ones = torch.ones(
```

Comment:
If it's ok by you, I prefer not sending things to numpy and cpu, and then back to torch, during training. we can do all the indexing and inverting in torch, no?
1. having this little nondifferentiable block might have a bad effect on gradients.
2. also might slow down training times?

Reply:
none of the transforms should be differentiable, these are fixed parameters; but i can put everything in torch anyways for consistency